sql/dd/impl/sdi.cc

```diff
@@ -380,8 +380,13 @@ bool CheckDefaultCompatibility(const RJ_Document &doc) {
 
   const RJ_Value &mysqld_version_id = doc["mysqld_version_id"];
   assert(mysqld_version_id.IsUint64());
-  if (mysqld_version_id.GetUint64() > std::uint64_t(MYSQL_VERSION_ID)) {
+  if (mysqld_version_id.GetUint64() > std::uint64_t(MYSQL_VERSION_ID) &&
+      ((mysqld_version_id.GetUint64() < std::uint64_t(221003)) ||
+       (mysqld_version_id.GetUint64() >= std::uint64_t(221100)))) {
     // Cannot deserialize SDIs from newer versions.
+    // RonDB 22.10.3 is assumed compatible with new versions of the
```

Comment:
This is a comment about *not* taking this branch, so perhaps the comment shouldn't be inside the if block. Maybe move to inside the condition?

Reply:
Ignored this comment